Code/Framework/AzFramework/AzFramework/Process/ProcessCommunicatorTracePrinter.cpp

```diff
@@ -80,7 +84,10 @@ void ProcessCommunicatorTracePrinter::WriteCurrentString(bool isFromStdErr)
         }
         else
         {
-            AZ_TracePrintf(m_window.c_str(), "%s", bufferToUse.c_str());
+            // Note that the input string will never have a newline at the end of it
+            // because it comes from ParseDataBuffer above, which splits by newline
+            // but does not include the newlines.
+            AZ_TracePrintf(m_window.c_str(), "%s\n", bufferToUse.c_str());
```

Comment:
We could skip format string by using `AZ::Debug::Trace::Output` directly.
That also has the benefit of avoiding the 4096 character limit of the [AZ::Debug::Trace::Printf](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Debug/Trace.cpp#L587) function.
```suggestion
            AZ::Debug::Trace::Output(m_window.c_str(), bufferToUse.c_str());
            AZ::Debug::Trace::Output(m_window.c_str(), "\n");
```

Reply:
I'l check but I think this is just a widely used traceprint "mirror" that reflects stdout back into the trace system, which means some of the listeners might be listening to onprint, not onOutput...

Reply:
I am just thinking of the case if for some reason more than 4096 bytes of output comes from the builder process before it gets pumped out.
It could truncate the output.

Reply:
Yes, that could happen (although not in practice in builders and AP since on the other side of this, in the builder, the one that is emitting the stdout/stderr is doing so via printf anyway and thus already has the limit).    I'll take a look at the code today and see if anything is listening to OnError, OnPrintf, OnWarning instead of OnOutput, if they are, I can't really convert this to OnOutput, since it bypasses all of those.

Reply:
I checked, both the MultiplayerEditorComponent and BuilderManager listen to traceprints and error and warning and so on, not output.  So It'd be a bad idea to move this to output.